applications/poisson/sine/application.h

```diff
@@ -216,9 +217,44 @@ class Application : public ApplicationBase<dim, n_components, Number>
       }
     };
 
-    this->grid->create_triangulation(this->param.grid,
-                                     lambda_create_coarse_triangulation,
-                                     this->param.grid.n_refine_global);
+    auto const lambda_read_external_coarse_grid = [&](dealii::Triangulation<dim, dim> & tria) {
+      AssertThrow(!this->param.grid.grid_file.empty(),
+                  dealii::ExcMessage(
+                    "You are trying to read a grid file, but the the string is empty."));
+
+      dealii::GridIn<dim> grid_in;
+
+      grid_in.attach_triangulation(tria);
+
+      std::string extension =
+        this->param.grid.grid_file.substr(this->param.grid.grid_file.find_last_of('.') + 1);
```

Comment:
I think we should think about the name `grid_file`. The above line shows us that we get expressions like `grid.grid...`. So I suggest either `filename` or `file` to avoid this.

Reply:
Yes, but `grid` could potentially have more external files in the future, for example an external mapping, see: https://github.com/exadg/exadg/pull/323#issuecomment-1407440353. That is why I decided for `grid_file` to distinguish this from a future `mapping_file`. Both `file` and `filename` were to unspecific for me.

What do you think about `triangulation_file`?

Reply:
According to my experience in software development, I would introduce a more complicated name only when needed. E.g. currently we do not know whether these mapping files become general ExaDG functionality, or are application specific, ... .

Reply:
changed in d932484523cc4b0a6ccbe41963e79a51e5d74191.